WcaOnRails/app/models/abstract_cached_model.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class AbstractCachedModel < ActiveRecord::Base
```

Comment:
OK, I think we now understand each other, and mostly have some reasonable differences of opinion. This is progress!

I think your new `cfind()` can be simplified:

    def self.cfind(id)
      class_variable_set(:@@cache, {}) unless class_variable_defined?(:@@cache)
      class_variable_get(:@@cache)[id] ||= find(id)
    end

I'm OK with the readability of that.

Still, having an abstract base class is a very non Ruby way of doing things. I think/hope this could just as well be done as a Module?

Do we really need to cache `.all()`? Of course we could, but is `.all()` a relevant part of the performance problems we're trying to solve?

If we do need it, things do get more complicated... I can imagine several unsatisfactory schemes :)